guides/common/modules/proc_decreasing-performance-impact-of-the-pull-based-rex-transport.adoc

```diff
@@ -0,0 +1,31 @@
+[id="Decreasing_performance_impact_of_the_pull-based_REX_transport_{context}"]
+= Decreasing Performance Impact of the Pull-Based REX Transport
+
+When {ProjectServer} is configured to use REX pull-based transport for remote execution jobs using "Script" provider, {SmartProxyServer} sends a notification about new job to the client machines via MQTT.
+This notification does not include actual workload (actual script) the client is supposed to execute.
+Once a client receives a message about a new remote execution job, it queries the {SmartProxyServer} for actual payload.
+As the client executes the job, it periodically sends outputs of the job to the capsule,further increasing the number of requests to {SmartProxyServer}.
+
+These requests to {SmartProxyServer} together with high concurrency allowed by MQTT protocol can cause exhaustion of {SmartProxyServer} available connections and some requests might fail, making some child tasks of remote execution jobs to hang. 
+This also depends on actual job payload, as some jobs are causing additional load to {ProjectServer}, making it compete for resources if you are using integrated {SmartProxyServer}.
+
+To avoid this, configure your {ProjectServer} and {SmartProxyServer} deployment appropriately with the following parameters:
+
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+* MQTT Time To Live {endash} Time interval in seconds given to the host to pick up the job before considering the job undelivered
+* MQTT Rate Limit {endash} Number of jobs that are allowed to run at the same time
+* MQTT Resend Interval {endash} Time interval in seconds at which the notification should be re-sent to the host until the job is picked up or cancelled
+----
+In order to tune these parameters you can follow the following procedure on the {ProjectServer}:
+
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+{foreman-installer} \
+--foreman-proxy-plugin-remote-execution-script-mqtt-rate-limit _My_MQTT_Rate_Limit_ \
+--foreman-proxy-plugin-remote-execution-script-mqtt-resend-interval _My_MQTT_Resend_Interval_ \
+--foreman-proxy-plugin-remote-execution-script-mqtt-ttl _My_MQTT_Time_To_Live_
+----
```

Comment:
Do you know of any sensible values a user might pick? @Imaanpreet

Reply:
Hello @maximiliankolb : I am doing the performance testing in order to give you some values, will share when we are done with our testing.

Reply:
Hello @adamruzicka : can we suggest some values from our testing now?

Reply:
Added the content. So resolving this conversation.